packages/emitter-framework/test/typescript/components/value-expression.test.tsx

```diff
@@ -0,0 +1,178 @@
+import { Output, render } from "@alloy-js/core";
+import { SourceFile } from "@alloy-js/typescript";
+import { ArrayValue, EnumValue, Namespace, Value } from "@typespec/compiler";
+import { $ } from "@typespec/compiler/experimental/typekit";
+import { format } from "prettier";
+import { assert, describe, expect, it } from "vitest";
+import { ValueExpression } from "../../../src/typescript/components/value-expression.js";
+import { getProgram } from "../test-host.js";
+
+describe("ValueExpression", () => {
+  async function testValueExpression(value: Value, expected: string) {
+    const res = render(
+      <Output>
+        <SourceFile path="test.ts">
+          const val = <ValueExpression value={value} />
+        </SourceFile>
+      </Output>,
+    );
+    const testFile = res.contents.find((file) => file.path === "test.ts");
+
+    assert.exists(testFile, "test.ts file not rendered");
+    const actualContent = await format(testFile.contents as string, { parser: "typescript" });
+    const expectedContent = await format(`const val = ${expected}`, {
+      parser: "typescript",
+    });
+    expect(actualContent).toBe(expectedContent);
+  }
+
+  it("handles string value", async () => {
+    // Error: Default typekits may not be used until a program is set in the compiler.
+    // TODO: better way to set this up so I can use typekit?
+    await getProgram(``);
```

Comment:
can this be run once on beforeAll?

Reply:
huh I could have sworn I tried that and it did not work. But I think I was bitten by the `vitest::beforeEach` vs `node::beforeEach` (VSCode will sometimes import the latter which does not actually run / work and its easy to miss). I Will use beforeAll if that works 👍